src/analysis/retail/druid/restoration/modules/spells/BuddingLeaves.tsx

```diff
@@ -0,0 +1,101 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { Options } from 'parser/core/Module';
+import { TALENTS_DRUID } from 'common/TALENTS';
+import Events, { HealEvent, RemoveBuffEvent } from 'parser/core/Events';
+import { lifebloomSpell } from 'analysis/retail/druid/restoration/constants';
+import SPELLS from 'common/SPELLS';
+import { encodeEventTargetString } from 'parser/shared/modules/Enemies';
+import { calculateEffectiveHealing } from 'parser/core/EventCalculateLib';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import ItemPercentHealingDone from 'parser/ui/ItemPercentHealingDone';
+
+const DEBUG = false;
+
+const HOT_BOOST_PER_RANK_PER_STACK = 0.03;
+const MAX_STACKS = 15;
+const BLOOM_BOOST_PER_RANK = 0.075;
+
+/**
+ * **Budding Leaves**
+ * Spec Talent
+ *
+ * Lifebloom's healing is increased by (3 / 6)% each time it heals, up to (45 / 90)%.
+ * Also increases Lifebloom's final bloom amount by (7.5 / 15)%.
+ * ------------------------------------
+ * TODO bugged? on live to cap out at 37/75% - remove this note if they fix it and change numbers if they don't
+ */
+export default class BuddingLeaves extends Analyzer {
+  ranks: number;
+
+  /** Mapping from encoded target ID to number of LB boost stacks they have */
+  stacksByTarget: Map<string, number> = new Map<string, number>();
+
+  /** Total healing attributable to this talent */
+  totalHealing: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.ranks = this.selectedCombatant.getTalentRank(TALENTS_DRUID.BUDDING_LEAVES_TALENT);
+    this.active = this.ranks > 0;
```

Comment:
Can you add a return after the active check?
```suggestion
    this.active = this.ranks > 0;
    if (!this.active) {
      return;
    }
```

Reply:
Not actually needed because the event subscribe doesn't happen for inactive modules (this is why you won't see the return anywhere else)

Here is the implementation of Module's addEventListener:
```ts
addEventListener<ET extends EventType, E extends MappedEvent<ET>>(
    eventFilter: ET | EventFilter<ET>,
    listener: EventListener<ET, E>,
  ) {
    if (!this.active) {
      return;
    }
    this.owner.addEventListener(eventFilter, listener.bind(this), this);
  }
```

Reply:
I've been living a lie. Thanks!

Reply:
It's super hidden behavior, I acknowledge that 😅